app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -347,14 +347,22 @@ def ca_cert_store(cert)
       raise _("Failed to create X509 certificate, error: %s" % e.message)
     end
 
-    def ca_cert
+    def fetch_unverified(path, query = '')
       ca_url = URI.parse(url)
-      ca_url.path = "/ca.crt"
+      ca_url.path = path
+      ca_url.query = query
       http = Net::HTTP.new(ca_url.host, ca_url.port)
       http.use_ssl = (ca_url.scheme == 'https')
       http.verify_mode = OpenSSL::SSL::VERIFY_NONE
-      request = Net::HTTP::Get.new(ca_url.path)
+      request = Net::HTTP::Get.new(ca_url)
       http.request(request).body
+    rescue Exception => e
```

Comment:
because this catches also exceptions like SyntaxError, SystemExit, NoMemoryError etc... you probably wanted to rescue StandardError or just do `rescue => e` which is equivalent